soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4081,6 +2334,9 @@ void RandomizerSettingsWindow::DrawElement() {
         UIWidgets::ReEnableComponent("");
     }
     ImGui::End();
+    if (changed) {
+        ctx->GetSettings()->UpdateOptionProperties();
+    }
```

Comment:
I'm now realizing that `RandomizerSettingsWindow` has an unused `UpdateElement` function which could probably be used here, instead of calling this function (responsible for updating Disabled/Hidden and certain gui flags). That might be an easy enough change to get into this PR, will check.

Reply:
I actually just pushed a commit to address this. `RandomizerSettingsWindow` now has a property for keeping track of whether or not it needs an update, which `UpdateElement` then checks and runs the `Settings::UpdateOptionProperties` function.